### PR TITLE
Issue #45 #47 #48 : passport

### PR DESCRIPTION
**Pull request by LChoiSH**

### CODE REVIEW

File: Backend/app.js [hunk 1]
```diff
@@ -1,19 +1,29 @@
...
 }));
 
+const passport = require('./lib/passport')(app);
+const rootRouter = require('./router/index')(passport);
+
 app.use('/api', rootRouter);
 
-app.listen(3000, () => {
-  console.log("Server Start");
+app.listen(80, () => {
```
Comment (minoring) [inline Backend/app.js]: I think hard coding port number is not a good idea.
How about making a constant variable?
like const port = (process.env.PORT || 80);

File: Backend/router/v1/user.js [hunk 2]
```diff
@@ -1,98 +1,85 @@
...
-      }
+  router.get('/signout', (req, res) => {
+    if(req.isAuthenticated()){
+      res.status(200);
+      req.session.destroy();
+      res.clearCookie('sid');
+      res.json({
+        msg: "signout success"
+      });
+=======
```
Comment (minoring) [inline Backend/router/v1/user.js]: =====

Comment (mtnviewlab) [reply Backend/router/v1/user.js]: Looks like this is the leftover merge conflict resolution.
Have you guys ever run the app after you merge?

File: Backend/router/v1/report.ctrl.js [hunk 3]
```diff
@@ -1,51 +1,69 @@
...
+        userID: req.user[0].dataValues.id
+      }
+    }).then((result) => {
+      res.status(200);
+      res.json(result);
+    }).catch((err) => {});
+  } else{
+    res.status(401)
+    res.json({
+      msg: "get report failed"
```
Comment (shberry) [inline Backend/router/v1/report.ctrl.js]: This is retrieving all reports, right? If so, 'Failed to get all reports' or something along this line might be better.

File: Backend/router/v1/report.ctrl.js [hunk 4]
```diff
@@ -55,15 +73,49 @@ const deleteReport = (req, res) => {
...
-      res.json({msg: "ReportId not exist"});
-    } else {
-      res.status(204).json(result);  //no content
-    };
-  }).catch((err) => {
-    res.json({msg: "error"});
-  });
-}
+    res.status(204).json(result);  //no content
+  }).catch((err) => {});
```
Comment (shberry) [inline Backend/router/v1/report.ctrl.js]: Are we intentionally ignoring the error case and not doing anything? If so, please add a comment to justify your decision.

File: Backend/router/v1/report.ctrl.js [hunk 5]
```diff
@@ -55,15 +73,49 @@ const deleteReport = (req, res) => {
...
+    return res.status(400).end()
+  }
+
+  const newWhat = req.body.data[0].what;
+  const newLocation = req.body.data[0].location;
+  const newTime = req.body.data[0].time;
+  const newWho = req.body.data[0].who;
+  const newDetails = req.body.data[0].details;
+
+  if(req.isAuthenticated()){
```
Comment (shberry) [inline Backend/router/v1/report.ctrl.js]: if (...) {

File: Backend/router/v1/report.ctrl.js (hunk 3 above)
Comment (eunyyyoung) [reply Backend/router/v1/report.ctrl.js]: I'm so sorry for the trouble, Soo. I modified it in my branch.

File: Backend/router/v1/report.ctrl.js (hunk 4 above)
Comment (eunyyyoung) [reply Backend/router/v1/report.ctrl.js]: I'm so sorry for the trouble, Soo. I modified it in my branch, too.

**Comment by eunyyyoung** [comment]
It's because we are working on the same code. So we try to fix own part. @mtnviewlab

**Comment by LChoiSH** [comment]
@mtnviewlab @shberry 
Could you review this code?

File: Backend/router/v1/report.js [hunk 6]
```diff
@@ -2,18 +2,20 @@ const express = require('express');
...
+module.exports = () => {
+  router.use((req, res, next) => {
+    next();
+  });
 
-router.all('/', (req, res) => {
-  res.send("this is report root");
-});
+  router.all('/', (req, res) => {
+    res.status(404).send("This is not normal route");
```
Comment (mtnviewlab) [inline Backend/router/v1/report.js]: Where do you send this "This is not normal route" message?
And how it gets displayed?

**Pull request merged**